python/ray/tune/trial_executor.py

```diff
@@ -165,6 +166,8 @@ def on_no_available_trials(self, trial_runner):
         if self._queue_trials:
             return
         for trial in trial_runner.get_trials():
+            if isinstance(trial.resources, PlacementGroupFactory):
+                return
```

Comment:
should this be a check before the loop has started?

Reply:
Unfortunately this is a trial property (even when not overloading `trial.resources`) so we can only check this in the loop. This should only be a problem when mixing non-pg trials and pg-trials (which we might want to check and explicitly not support)